apps/lib/apps.c

```diff
@@ -2370,41 +2366,58 @@ int raw_write_stdout(const void *buf, int siz)
  * does impact behavior on some platforms, such as differentiating between
  * text and binary input/output on non-Unix platforms
  */
+static BIO *check_bio(BIO *bp, const char *what, int mode)
+{
+    if (bp == NULL) {
+        BIO_printf(bio_err, "%s: Could not open %s in mode 0x%x\n",
+                opt_getprog(), what, mode);
+        if (bio_err != NULL)
+            ERR_print_errors(bio_err);
+        exit(1);
+    }
+    return bp;
+}
+
 BIO *dup_bio_in(int format)
 {
-    return BIO_new_fp(stdin,
-                      BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0);
+
+    return check_bio(BIO_new_fp(stdin, mode), "stdin", mode);
 }
 
 BIO *dup_bio_out(int format)
 {
-    BIO *b = BIO_new_fp(stdout,
-                        BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0);
+    BIO *b = BIO_new_fp(stdout, mode);
     void *prefix = NULL;
 
+    /* Only care about NULL error-exit. */
+    check_bio(b, "stdout", mode);
+
 #ifdef OPENSSL_SYS_VMS
     if (FMT_istext(format))
         b = BIO_push(BIO_new(BIO_f_linebuffer()), b);
 #endif
 
     if (FMT_istext(format)
-        && (prefix = getenv("HARNESS_OSSL_PREFIX")) != NULL) {
+            && (prefix = getenv("HARNESS_OSSL_PREFIX")) != NULL) {
         b = BIO_push(BIO_new(BIO_f_prefix()), b);
         BIO_set_prefix(b, prefix);
     }
 
-    return b;
+    return check_bio(b, "stdout", mode);
 }
 
 BIO *dup_bio_err(int format)
 {
-    BIO *b = BIO_new_fp(stderr,
-                        BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0);
+    BIO *b = BIO_new_fp(stderr, mode);
```

Comment:
likewise checks missing here

Reply:
same reason, if you push a NULL bio you get the original BIO back so no check needed.

Reply:
I think the check is missing here.

Reply:
the check_bio is still missing here.

Reply:
(or in th SYS_VMS ifdef block)

Reply:
forgot to push.